mesonbuild/wrap/wrap.py

```diff
@@ -66,16 +67,23 @@ def whitelist_wrapdb(urlstr: str) -> urllib.parse.ParseResult:
         raise WrapException(f'WrapDB did not have expected SSL https url, instead got {urlstr}')
     return url
 
-def open_wrapdburl(urlstring: str, allow_insecure: bool = False, have_opt: bool = False) -> 'http.client.HTTPResponse':
+def open_wrapdburl(urlstring: str, allow_insecure: bool = False, have_opt: bool = False, allow_compression: bool = False) -> 'http.client.HTTPResponse':
     if have_opt:
         insecure_msg = '\n\n    To allow connecting anyway, pass `--allow-insecure`.'
     else:
         insecure_msg = ''
 
+    def do_urlopen(url: urllib.parse.ParseResult) -> 'http.client.HTTPResponse':
+        headers = {}
+        if allow_compression:
+            headers['Accept-Encoding'] = 'gzip'
```

Comment:
Why not unconditionally allow this (or at least, enable it by default, you're passing True to every call of this function anyway)? If the server doesn't support it, it can just not send compressed data.

Reply:
Callers that enable `allow_compression` also need to read the response body with `read_and_decompress()` or they risk writing gzipped files to disk (or e.g. trying to parse them as text) so it seems better to require the caller to explicitly ask for it.  Not every call site wants this.  `get_data()` doesn't enable compression because a) it wants to read in chunks, which `read_and_decompress()` doesn't bother to support, and b) the resource it's fetching is likely already compressed.